python/ray/data/dataset.py

```diff
@@ -3975,9 +3973,7 @@ def to_tf(
             prefetch_batches: The number of batches to fetch ahead of the current batch
                 to fetch. If set to greater than 0, a separate threadpool is used
                 to fetch the objects to the local node, format the batches, and apply
-                the collate_fn. Defaults to 1. You can revert back to the old
-                prefetching behavior that uses `prefetch_blocks` by setting
-                `use_legacy_iter_batches` to True in the :class:`~ray.data.DataContext`.
```

Comment:
Also remove the deprecated prefetch_blocks args from all these methods, and the ones in DataIterator

Reply:
`DatasetPipeline` has methods that depend on `prefetch_blocks`. To minimize code conflicts, I'm planning on sequencing the deprecations like this:
1. Remove legacy iteration code path 
2. Remove `DatasetPipeline`
3. Remove `prefetch_blocks`